engine/src/main/java/org/hibernate/validator/internal/util/TypeHelper.java

```diff
@@ -324,24 +326,32 @@ private static Type getResolvedSuperclass(Type type) {
 		return validatorsTypes;
 	}
 
-	public static Type extractType(Class<? extends ConstraintValidator<?, ?>> validator) {
-		Map<Type, Type> resolvedTypes = newHashMap();
+	public static Type extractConstraintValidatorValidatedType(Class<? extends ConstraintValidator<?, ?>> validator) {
```

Comment:
How about `extractValidatedType()` and `extractConstraintType()`? The context is clear from the parameter type and I had to read twice to see where the two method names are different.

Reply:
Frankly, I don't like having such specific methods in the very generic `TypeHelper` (I opened a new issue for that), that's why I was over cautious. Can shorten the name though.